docs/docs/api/Dispatcher.md

```diff
@@ -817,6 +817,131 @@ try {
 }
 ```
 
+### `Dispatcher.compose(interceptors[, interceptor])`
+
+Compose a new dispatcher from the current dispatcher and the given interceptors.
+
+> _Notes_:
+> - The order of the interceptors is important. The first interceptor will be the first to be called.
+> - It is important to note that the `interceptor` function should return a `Dispatcher` instance.
+> - Any fork of the chain of `interceptors` can lead to unexpected results, it is important that an interceptor returns a `Dispatcher` instance that forwards the request to the next interceptor in the chain.
+
+Arguments:
+
+* **interceptors** `Interceptor[]`: It is an array of `Interceptor` functions passed as only argument, or several interceptors passed as separate arguments.
+
+Returns: `Dispatcher`.
+
+#### Parameter: `Interceptor`
+
+A function that takes a `Dispatcher` instance and returns a `Dispatcher` instance.
+
+#### Example 1 - Basic Compose
```

Comment:
Why would somebody use this while we provide a the `RedirectDispatcher` ourselves? Let's work on how people would use this API.

Reply:
I mean the point is to make it easy to compose these kinds of "interceptor" dispachers, i.e. dispatchers that take another dispatcher + options in its constructor.